zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreter.java

```diff
@@ -324,18 +324,19 @@ public void setInterpreterGroup(InterpreterGroup interpreterGroup) {
     super.setInterpreterGroup(interpreterGroup);
 
     synchronized (interpreterGroupReference) {
-      if (interpreterGroupReference
-          .containsKey(getInterpreterGroupKey(interpreterGroup))) {
-        interpreterGroupReference.remove(getInterpreterGroupKey(interpreterGroup));
+      if (!interpreterGroupReference
+          .containsKey(getInterpreterGroupKey(interpreterGroup))
+          || (!interpreterGroupReference.get(
+              getInterpreterGroupKey(interpreterGroup)).isRunning() && interpreterGroupReference
+              .get(getInterpreterGroupKey(interpreterGroup)).getPort() > 0)) {
+        interpreterGroupReference.put(getInterpreterGroupKey(interpreterGroup),
```

Comment:
nit: instead of ||, make it nested such that you can set `interpreterGroupReference.get(getInterpreterGroupKey(interpreterGroup))`
into a variable and reuse it instead of calling it multiple times? This probably doesn't have any concurrency/synchronization/thread-safety issue? So this is more a nit